pkg/tuned/tuned.js

```diff
@@ -0,0 +1,178 @@
+    "base1/mustache",
+    "base1/patterns"
+], function($, cockpit, po, Mustache) {
+    "use strict";
+
+    cockpit.locale(po);
+    cockpit.translate();
+    var _ = cockpit.gettext;
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "tuned")
+            console.debug.apply(console, arguments);
+    }
+
+    /*
+     * gather and display information regarding tuned profiles
+     *
+     * update_info: if true, trigger an immediate update of the information
+     */
+    function tuned(update_info) {
+        var service = cockpit.dbus('com.redhat.tuned');
+        var profile;
+
+        /* we trigger status update via dbus
+         * if we don't get a timely reply, tuned is probably not running
+         */
+        var update_timeout;
+
+        /* The promise for the last operation */
+        var last_promise;
+
+        /* Tuned doesn't provide GetAll(), so we can't introspect */
+
+        //TODO: implement fail() and other bits to make this nicer
+        function wrapper(method, args, cb) {
+            if (args === null)
+                args = []
+            var call = service.call(
+                '/Tuned',
+                'com.redhat.tuned.control',
+                method,
+                args).
+                done(function(result) { cb(result);});
+        }
+
+        function get_profiles() {
+            wrapper('profiles', [], function(result) {
+                var profiles = result.toString().split(',');
+                $('#change-tuned').empty();
+                for (var i = 0; i < profiles.length; i++) {
+                    $('#change-tuned').append($('<option>', {
+                        value: profiles[i],
+                        text: profiles[i],
+                        selected: profiles[i] === profile
+                    }));
+                }
+                $('.selectpicker').selectpicker('refresh');
+            });
+        }
+
+        function get_profile() {
+            var call = service.call(
+                '/Tuned',
+                'com.redhat.tuned.control',
+                'active_profile',
+                []).
+                    done(function(result) {
+                        update_timeout = null,
+                        $('#tuned-profile-button').text(result);
+                        profile = result[0];
+                        $('#tuned-system-status').show();
+                    });
+            update_timeout = window.setTimeout(status_update_failed, 10000);
+        }
+
+        function status_update_failed() {
+            if (update_timeout !== null) {
+                $('#tuned-not-found').show();
+            }
+            update_timeout = null;
+        }
+
+        function remove_notifications() {
+            $('div.container-fluid.alert').remove();
+        }
+
+        $('#tuned-start').on('click', function() {
+            var proc = cockpit.spawn(["systemctl", "restart", "tuned.service"]).
```

Comment:
Recently @mvollmer has added a handy ```system/service``` wrapper javascript API to accomplish this task. You can see it being used in ```docker/containers.js```.

I think broadly we would do something like:

```
define([
...
    "system/service",
], function(..., service, ...) {
...
var tuned_service = service.proxy("tuned");
tuned_service.start().
      done(function () { ... }).
      fail(function (error) {
            show_curtain(cockpit.format(_("Failed to start Tuned: $0"), error));
      });
```